src/CosmosEditorManager.ts

```diff
@@ -54,7 +58,7 @@ export class CosmosEditorManager {
         this._globalState.update(this._persistedEditorsKey, fileMapLabels);
         const textEditor = await vscode.window.showTextDocument(document);
         const data = await editor.getData();
-        await this.updateEditor(data, textEditor);
+        await this.updateEditor(data, textEditor, editor);
```

Comment:
This is why I don't like the name ICosmosEditor, it's very confusing.  But don't worry about it now.